### PR TITLE
Added artists manually and fetching their songs 

### PR DESCRIPTION
**Pull request by RahulRudra81**
Related Issue: #376

1. ### **Firebase Integration:**

- Add 10 Artists Manually: Added 10 artists to the Firebase database, each with relevant details such as name and image.
- Fetch Artist Data: Implemented functionality to fetch the names and images of these artists from Firebase, ensuring that the data is dynamically loaded into the application.
- Also added one more field in the user data called img where you can put image of the artists.

image

2. ### **Enhanced Artist Interaction:**

- Artist Click Event: Users can now click on an artist's name or image in the 'Top Artists' section.
- Dynamic Top Songs: Upon clicking an artist, the 'Top Songs' section updates to display the top songs of the selected artist, providing a more interactive and personalized experience.
- Playable Top Songs: Implemented functionality to play the songs listed in the 'Top Songs' section directly, allowing users to listen to their favorite tracks easily.

### Changes I have made

1. ### First time when you open the website you can see the list of the artists.

image

2. ### When you click on any artist the top songs will be replaced by the top songs of the selected artist.

image

3. ### When you click on any song it plays now .
image

### CODE REVIEW

**Comment by Satyam1923** [comment]
Screenshot 2024-06-17 162504
@RahulRudra81 did you create an instance for firebase app because it shows an error for permissions.

**Comment by Satyam1923** [comment]
Screenshot 2024-06-17 162504
which permissions is it requesting.

**Comment by RahulRudra81** [comment]
it cant access any database because ig its deployed on your vercel acc . @Satyam1923

**Comment by RahulRudra81** [comment]
@Satyam1923  go to the deployment above and add firebase credentials . i think then it should work

**Comment by Satyam1923** [comment]
Screenshot 2024-06-17 164235
the problem is untill you are logged in it won't let you read data from firebase and it's still not working may be because one more field image in it

**Comment by Satyam1923** [comment]
@RahulRudra81 don't do this Firebase thing I'll provide you just a  backend endpoint use it for top artists.

**Comment by RahulRudra81** [comment]
@Satyam1923 ok sure

**Comment by Satyam1923** [comment]
Screenshot 2024-06-17 190049
this type of data you will get from backend.

**Comment by RahulRudra81** [comment]
han will work for me no problem . provide me with the endpoint api.

**Comment by Satyam1923** [comment]
@RahulRudra81 done now you can use endpoints for data.

**Comment by RahulRudra81** [comment]
Which endpoint return me this data?

**Comment by Satyam1923** [comment]
@RahulRudra81 hit this endpoint /top-artists

**Comment by RahulRudra81** [comment]
I actually cant run the backend i guess there is some env file for backend also

**Comment by Satyam1923** [comment]
@RahulRudra81 your master branch is behind 9 commits. Pull new changes then it won't have env.

**Comment by RahulRudra81** [comment]
I have pulled the branch and still getting this error when I start the backend 
image
The error is right here in scheduler.js 
image

**Comment by Satyam1923** [comment]
@RahulRudra81 now pull fixed the master branch.

**Comment by RahulRudra81** [comment]
oh yes its working

**Comment by RahulRudra81** [comment]
Can you tell in which file is that /top-Artist route written cause I am not able to find it in index .js nor can hit with postman
@Satyam1923

**Comment by RahulRudra81** [comment]
@Satyam1923 I couldn't find the route so I created one and now I am able to fetch so I will push along with the in frontend

**Comment by RahulRudra81** [comment]
@Satyam1923 i have put fetch request to http://localhost:3030/top-Artists how will you see the deployed changed?

**Comment by Satyam1923** [comment]
https://spring-music-player-3hyj.vercel.app use this baseurl and /top-artists endpoint and pull new changes.

**Comment by RahulRudra81** [comment]
oh ok wait i will do

File: backend/index.js [hunk 1]
```diff
@@ -44,8 +44,18 @@ app.use("/search/artists", getArtists);
 //get artist by id
 app.use("/search/artist", getArtistsById);
 
-//get top artists
-app.use("/top-artists",getTopArtists)
+
+app.use("/top-Artists",getTopArtists)
```
Comment (Satyam1923) [inline backend/index.js:48]: now remove these.

File: frontend/src/Utils.js [hunk 2]
```diff
@@ -1,4 +1,7 @@
 import axios from "axios";
+import { collection, getDocs } from 'firebase/firestore';
+import { db } from './components/Auth/firebase';
```
Comment (Satyam1923) [inline frontend/src/Utils.js:3]: these also not needed.

File: frontend/src/Utils.js [hunk 3]
```diff
@@ -38,3 +41,11 @@ export const fetchTopSongs = async (setTopSongs) => {
     console.error("There was a problem with the fetch operation:", error);
   }
 };
+
+
```
Comment (Satyam1923) [inline frontend/src/Utils.js:45]: test also not needed.

File: backend/index.js [hunk 4]
```diff
@@ -44,8 +44,18 @@ app.use("/search/artists", getArtists);
...
+
```
Comment (Satyam1923) [inline backend/index.js:49]: remove this too.

File: frontend/src/components/Home/Home.jsx [hunk 5]
```diff
@@ -172,61 +176,104 @@ function TopChartsElement({ songName, artistName, songDuration }) {
...
   );
 }
 
-function TopArtists() {
+function TopArtists({ setCurrentArtist }) {
+  const [artists, setArtists] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await fetch("http://localhost:3030/top-Artists")
```
Comment (Satyam1923) [inline frontend/src/components/Home/Home.jsx:244]: change this baseurl

**Comment by RahulRudra81** [comment]
Wait i will resolve and push again my bad

**Comment by Satyam1923** [comment]
@RahulRudra81 could please styling for it looks too bad.

**Comment by RahulRudra81** [comment]
image

does this look better?

**Comment by Satyam1923** [comment]
Yaa now looks fine.

**Comment by Satyam1923** [comment]
@RahulRudra81 you are doing it in linux right?

**Comment by RahulRudra81** [comment]
Yes

**Comment by Satyam1923** [comment]
@Satyam1923 zoom in I little bit or use 125% fractional scaling while fixing its CSS.

**Comment by RahulRudra81** [comment]
125% also it looks like this
image
normal 100% it look like this
image

**Comment by Satyam1923** [comment]
Screenshot 2024-06-18 115645
check here

**Comment by RahulRudra81** [comment]
bro the log-in button  position is fixed .

**Comment by Satyam1923** [comment]
@RahulRudra81 No it not fixed.

**Comment by Satyam1923** [comment]
Screenshot 2024-06-18 121243
Screenshot 2024-06-18 121238

also fix the numbers.

**Comment by Satyam1923** [comment]
Screenshot 2024-06-18 121332
reduce the spacing between recent played and top artist

**Comment by RahulRudra81** [comment]
Is this fine?
image
Also if you like it i can keep the scrollbar so user can scroll through the selected artist . See like this . If not I will remove scroll and commit.
[Screencast from 18-06-24 01:26:04 PM IST.webm](https://github.com/Satyam1923/Spring-Music-Player/assets/91683510/b890a1e3-ed4a-4f91-a52b-8a416afa527f)

**Comment by Satyam1923** [comment]
@RahulRudra81 ya looks fine.

**Pull request merged**